### ISSUE TITLE
Fatal error while opening courses in Snap theme

### ISSUE THREAD

**Issue by amita435** [open-lms-open-source/moodle-theme_snap#285]
Fatal error: Declaration of theme_snap\output\core\course_renderer::course_section_cm_completion($course, &$completioninfo, cm_info $mod, $displayoptions = []) must be compatible with core_course_renderer::course_section_cm_completion() in /var/www/moodle_1click/theme/snap/classes/output/core/course_renderer.php on line 29

The error comes while debugging.

Please help. New to Moodle.

**Comment by EdTechSavvy** [comment]
I am having the same problem with Snap theme PHP 8.1 Moodle version 4.3. 

Snap is using a method (core_course_renderer::course_section_cm_completion) that has been removed from core Moodle in 4.3+
From Moodle documentation:

=== 4.3 ===
* The core_course_renderer::course_section_cm_completion method has been removed, and can no longer be used
* External function core_course_external::get_course_contents() now returns a new field activitybadge with the data to display
  the activity badge when the module implements it.
* prepare_new_moduleinfo_data() now accepts a parameter "suffix" that will be added to the name of the completion rules.
* The method core_course_bulk_activity_completion_renderer:: edit_default_completion() has been deprecated and will be removed.
* The core_course\output\activity_information output class has been deprecated. Use core_course\output\activity_completion
  and core_course\output\activity_dates instead.
  
Alongside with that, the following methods have been deprecated too:
  - core_course_renderer::render_activity_information()
  - renderer_base::activity_information()

**Comment by lvidgen** [comment]
Yes, this is broken and needs to be fixed by the dev team. Snap 4.3 will not work with Moodle 4.3 as reported in [Issue 284](https://github.com/open-lms-open-source/moodle-theme_snap/issues/284)
